### PR TITLE
feat: legal hold - update banners to v2 [WPB-6464]

### PR DESCRIPTION
**Pull request by saleniuk**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6464" title="WPB-6464" target="_blank">TaskWPB-6464</a>  [Android] Update banners to v2 according to designs 
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Designs for legal hold banners on “home” screen got updated, currently V2 is only available, so the app needs to be updated according to the recent designs.

### Solutions

Removed top bar legal hold banner which was visible on all logged in screens right below the call or connection banner.
Instead, added a border around the avatar which indicates that the user is under legal hold or has pending request. 
The dialog to approve it is available on the self user profile screen by clicking on “approve” on the legal hold pending banner. 
Used updated use case for observing self user legal hold state that includes legal hold pending request.

### Dependencies (Optional)

Needs releases with:

- https://github.com/wireapp/kalium/pull/2479

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

Enable legal hold or request it for the user.

### Attachments (Optional)

| Before | After |
| ----------- | ------------ |
| ezgif-frame-010 | Screenshot_20240206_172141 |

image

image

https://github.com/wireapp/wire-android/assets/30429749/89815e73-83ac-4dd3-93b8-d4c3362a53a0

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2937](https://10.10.124.134/job/AR-build-pipeline/2937/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(8827346)(8827346)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(8827346)(8827346)-staging-compat.apk)

**Review by MohamadJaara — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/ui/legalhold/ObserveLegalHoldStatusForCurrentUserUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                                else -> LegalHoldUIState.None
+                            }
+                        }
+                    }
+                    else -> flowOf(LegalHoldUIState.None)
+                }
+            }
+            .distinctUntilChanged()
+    }
+}
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/ui/legalhold/ObserveLegalHoldStatusForCurrentUserUseCase.kt]: Maybe have it in kalium side with others uses cases ?

Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/legalhold/ObserveLegalHoldStatusForCurrentUserUseCase.kt]: Sure, you're right, that's more clean, so I created this PR for kalium: https://github.com/wireapp/kalium/pull/2479
Please take a look and review it

**Review by ohassine — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [2988](https://10.10.124.134/job/AR-build-pipeline/2988/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(8843274)(8843274)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(8843274)(8843274)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [2992](https://10.10.124.134/job/AR-build-pipeline/2992/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(8843683)(8843683)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(8843683)(8843683)-staging-compat.apk)

**Pull request merged**